customer-managed/aws/terraform/iam_redpanda_agent.tf

```diff
@@ -52,6 +52,7 @@ data "aws_iam_policy_document" "redpanda_agent1" {
       # https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsidentityandaccessmanagementiam.html
       "iam:ListPolicies",
       "iam:ListRoles",
+      "iam:GetRole",
```

Comment:
question: Why do we need to be able to get all roles? This block of code here is supposed to grant us access to get any role we ask the customer to create or that we create ourselves...

https://github.com/redpanda-data/cloud-examples/blob/main/customer-managed/aws/terraform/iam_redpanda_agent.tf#L354-L377

Reply:
Changed it using 

`arn:aws:iam::${local.aws_account_id}:role/aws-service-role/eks-nodegroup.amazonaws.com/AWSServiceRoleForAmazonEKSNodegroup` 

to help me solve this problem 

```
Error: creating EKS Node Group (redpanda-d1r5rghhdd3miq02qs50:redpanda-d1r5rghhdd3miq02qs50-util-20250715144739926200000007): operation error EKS: CreateNodegroup, https response error StatusCode: 400, RequestID    : 591a8763-0728-468d-9db7-78a894771370, InvalidRequestException: Failed to validate if SLR: AWSServiceRoleForAmazonEKSNodegroup already exists due to missing permissions for 'iam:GetRole'
```